node/src/Commands.ts

```diff
@@ -2383,17 +2397,36 @@ export function createXRead(
     keys_and_ids: Record<string, string>,
     options?: StreamReadOptions,
 ): command_request.Command {
-    const args: string[] = [];
+    let args: string[] = [];
 
     if (options) {
-        addReadOptions(options, args);
+        args = addReadOptions(options);
     }
 
-    addStreamsArgs(keys_and_ids, args);
+    args = args.concat(addStreamsArgs(keys_and_ids));
 
     return createCommand(RequestType.XRead, args);
 }
 
+/** @internal */
+export function createXReadGroup(
+    group: string,
+    consumer: string,
+    keys_and_ids: Record<string, string>,
+    options?: StreamReadGroupOptions,
+): command_request.Command {
+    let args: string[] = ["GROUP", group, consumer];
+
+    if (options) {
+        args = args.concat(addReadOptions(options));
```

Comment:
concat clones the entire array
```suggestion
        args.push(addReadOptions(options));
```